domain/application/service/service.go

```diff
@@ -543,6 +545,28 @@ func (s *WatchableService) WatchApplicationConfigHash(ctx context.Context, name
 	)
 }
 
+// WatchApplicationSettings watches for changes to the specified application's
+// settings.
+// This functions returns the following errors:
+// - [applicationerrors.ApplicationNotFound] if the application doesn't exist
+func (s *WatchableService) WatchApplicationSettings(ctx context.Context, name string) (watcher.NotifyWatcher, error) {
```

Comment:
Tiny, tiny nit. All other watchers in this file have the signature:
```suggestion
func (s *WatchableService) WatchApplicationSettings(ctx context.Context, appName string) (watcher.NotifyWatcher, error) {
```